### ISSUE TITLE
Feature Request: Discarding based on containers (Firefox)

### ISSUE THREAD

**Issue by justinmok** [rNeomy/auto-tab-discard#287]
tl;dr- (White|Black)list Firefox tabs from discarding by using containers as groups
 
### **Feature Request**
I propose adding two new configuration options to support containers in Firefox:
1. Whitelist - **Only** tabs belonging to these containers may be considered for discarding
2. Blacklist - Tabs belonging to these containers **may not** be considered for discarding (inverse of whitelist)

For context, Firefox users can use the built-in containers feature to isolate tabs from one another. So in a sense, tabs can be organized by the containers they belong in, similar to tab groups in Chrome.
![image](https://user-images.githubusercontent.com/32044262/166859757-fe258254-f76c-4003-a13c-9d2f8c394d57.png)
_Tabs can be created in different containers._

### **Use Case**
Firefox users like myself can benefit from this feature by **simplifying tab discards without involving URLs/hostnames.**

For example, I use a combination of Simple Tab Groups and containers to organize my tabs. My browser is configured so that existing and future tabs in the "quarantine" group are added to the "quarantine" container. 
With this feature, I can make sure that none of my other, possibly important, tabs are discarded.

### **Mockup**
Here's what an implementation of this feature could look like:
![2022-05-04_20-15](https://user-images.githubusercontent.com/32044262/166858848-f29330c0-5d6d-419d-b4c5-54aa8dde3561.png)
_Options configured to only discard default (uncontained) tabs and tabs belonging to the "Personal" container.
Additionally, the option to blacklist is disabled as a result._

Notes:
- The `contextualIdentities` [(MDN)](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/contextualIdentities) API can be used to retrieve tab container information
- ATD users will also need to enable the containers feature, even if they haven't configured a whitelist/blacklist. This is automated when the extension is installed, so I'm not sure if it's really a problem.
- To improve on the example above, the list of containers could be changed by appending each container's icon before the name, or perhaps by changing the text color to one of the containers'.

**Comment by mxmilkiib** [comment]
This would be great. My personal container can discard tabs whenever, and there are many many tabs in it that I slowly read and maybe document, but I really can't have any work container tab discard as there are often unsaved or uncompleted forms.
